keyResolver.js

```diff
@@ -1,16 +1,24 @@
 /*!
  * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
  */
-'use strict';
+import {httpClient} from '@digitalbazaar/http-client';
+import assert from 'assert-plus';
+import didIo from 'did-io';
+import {driver as keyDriver} from 'did-method-key';
+import {driver as veresDriver} from 'did-method-key';
 
-import axios from 'axios';
+// config did-io to support did:key and did:v1 drivers
+didIo.use('key', keyDriver());
+didIo.use('v1', veresDriver());
 
-// FIXME: make more restrictive, support `did:key` and `did:v1`
-// TODO: could be made more restrictive is based on a config option that
-//       specifies where the KMS is.
-async function keyResolver({id} = {}) {
+async function keyResolver({id, didMethod = 'key'} = {}) {
+  assert.string(id, 'id');
+  assert.string(didMethod, 'didMethod');
+  if(id.startsWith('did:')) {
+    return didIo.get({did: id});
+  }
   const headers = {Accept: 'application/ld+json, application/json'};
-  const response = await axios.get(id, {
+  const response = await httpClient.get(id, {
```

Comment:
So this is going to hit *any* http URL that is passed in. We need to restrict this to only hit a URL that begins with a certain prefix (the KMS system's prefix). I think, since this is a `bedrock-web-*` lib, we can check to ensure that `id` starts with: `location.origin`.

Reply:
Does anyone know if we have a need for this library to resolve keys from anywhere else?

Reply:
I don't think we currently fetch keys from anywhere except for `did:key` and `did:v1`, in any of our libs.

Reply:
@dmitrizagidulin -- There may be other users of this library that hit a local WebKMS system for key agreement keys -- I'm not sure. I think if we limit this to local (same origin) traffic, we'll be ok. Unless someone wants to do an audit to ensure we only *ever* to `did:key` and `did:v1` ... we should do the same origin check right now, IMO.